pino-opentelemetry-transport.js

```diff
@@ -0,0 +1,136 @@
+'use strict'
+
+const build = require('pino-abstract-transport')
+const { SonicBoom } = require('sonic-boom')
+const { once } = require('events')
+
+const DEFAULT_MESSAGE_KEY = 'msg'
+
+const ZEROS_FROM_MILLI_TO_NANO = '0'.repeat(6)
+
+/**
+ * @typedef {Object} CommonBindings
+ * @property {string=} msg
+ * @property {number=} level
+ * @property {number=} time
+ * @property {string=} hostname
+ * @property {number=} pid
+ *
+ * @typedef {Record<string, string | number | Object> & CommonBindings} Bindings
```

Comment:
Shouldn't this be a separate docblock? Wouldn't this `@typedef` conflict with the first `@typedef` in this block?

Reply:
This is in the same block to keep the related typedefs together.

It won't conflict: the Bindings extends the CommonBindings which is intentional.


The idea was to signify the difference between the usual Bindings from the unknown ones.

Reply:
In addition. The plan is to move everything to TypeScript soon so this will get deleted anyway.

Reply:
😬 that would result in me voting against adding it to the Pino organization. If that's not a goal, okay.

Reply:
Writing it in TypeScript? I assumed so, which was the reason why I originally wrote it in JS. 

I had a problem with generating types when using the original pino-abstract-transport so I had to [fork it](https://github.com/Vunovati/pino-abstract-transport?organization=Vunovati&organization=Vunovati). My fork changes how the types are exported enabling them to be picked up by TS compiler from JsDoc when generating the .d.ts file. I did not submit a PR since it did not seem worthwhile to disturb the waters over this. 

I had a talk outside of this PR with @simoneb where we concluded that it might be simpler to just use TS.

Reply:
I updated the PR to include the manually generated types like in pino-abstract-transport. So no TypeScript needed.

Reply:
Maybe fix pino-abstract-transport then?

Reply:
Sure, I have a fix ready. I'll create an issue on pino-abstract-trasport describing what the problem was and submit the PR.